drivers/misc/nordic/nrf_cpu.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/event_device.h>
+
+static atomic_t block_cpu_idle;
+
+struct shim_config {
+	const struct pm_event_device *event_device;
+};
+
+static void shim_request_latency(const struct device *dev, uint8_t event_state)
+{
+	ARG_UNUSED(dev);
+
+	atomic_set(&block_cpu_idle, event_state);
+}
+
+static int shim_init(const struct device *dev)
+{
+	const struct shim_config *config = dev->config;
+
+	pm_event_device_init(config->event_device);
+	return 0;
+}
+
+void idle_enter_hook(void)
+{
+	if (atomic_get(&block_cpu_idle)) {
+		return;
+	}
+
+	(void)arch_irq_lock();
+	k_cpu_idle();
```

Comment:
Is there more latency when waiting for an interruption than spinning in a while loop  with an atomic check ?

Reply:
depends on the platform, but for nrf chips yep, around 4us, spinning keeps the CPU clock running (and probably some other hardware) :)